src/Hazelcast.Net/CP/CpSubSystemSession.Requests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Hazelcast.Core;
+using Hazelcast.Protocol.Codecs;
+
+namespace Hazelcast.CP
+{
+    /// <summary>
+    /// Implements server side requests for CP Session
+    /// </summary>
+    internal partial class CpSubsystemSession
+    {
+        /// <summary>
+        /// Generates a cluster-wide unique thread id for the caller
+        /// </summary>
+        /// <param name="groupId"></param>
+        /// <returns></returns>
+        protected async Task<long> RequestGenerateThreadIdAsync(CPGroupId groupId)
+        {
+            var requestMessage = CPSessionGenerateThreadIdCodec.EncodeRequest(groupId);
+            var responseMessage = await Cluster.Messaging.SendAsync(requestMessage).CfAwait();
+            var response = CPSessionGenerateThreadIdCodec.DecodeResponse(responseMessage);
+            return response.Response;
+        }
+
+        /// <summary>
+        /// Creates a new session on the server
+        /// </summary>
+        /// <param name="groupId"></param>
+        /// <returns>CPSubsystemSessionState and hearbeat milliseconds</returns>
+        protected async Task<(CPSubsystemSessionState, long)> RequestNewSessionAsync(CPGroupId groupId)
+        {
+            var requestMessage = CPSessionCreateSessionCodec.EncodeRequest(groupId, Cluster.ClientId.ToString());
+            var responseMessage = await Cluster.Messaging.SendAsync(requestMessage).CfAwait();
+            var response = CPSessionCreateSessionCodec.DecodeResponse(requestMessage);
```

Comment:
You want to decode the `responseMessage` not the `requestMessage`.

Reply:
Fixed during testing :)